generate-index.js

```diff
@@ -0,0 +1,40 @@
+const {decode} = require('html-entities');
+const fs = require('fs');
+const mkdirp = require('mkdirp');
+const path = require('path');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+// set directories and paths
+const methodsDir = path.join('methods');
+const indexFile = path.join(methodsDir, 'index.json');
+
+// Process all method files and generate an index file
+(async () => {
```

Comment:
how / when should this function be called.

Reply:
> seems like it should happen in CI, after entry validation.

Yes, exactly, and if there is any difference to index.json, a commit should be added. This means that additions/modifications to methods/*.json will also include an update to methods/index.json if any fields change.

Reply:
I added a comment here... https://github.com/w3c/did-spec-registries/pull/361/files#diff-e2f4a6697573e3de56bdab8e709373e54fe1afe9ba9a9e62839fe4637ae99888R21

Having this split into so many PRs makes it difficult to see what is going on...

My guess is that we will merge them all, and the registry will be broken until someone can clean it up...

> Yes, exactly, and if there is any difference to index.json, a commit should be added.

You don't need to commit the index to git, its a generated file... committing generated files is a bad practice.

Instead, index should be git ignored, builds that fail to generate it should fail, and it should be impossible to merge a PR that would break the index, this will also ensure that PR reviews don't contain a massive generated file that is thrashing every time a new json file is added.

I'm happy to do that work, assuming we can agree that it will need to be done without a "lengthy review" given the registry will be broken until it happens.

Reply:
I moved your script to the CI PR and wired it to run ONLY when all entries are valid. 

I also added some documentation about it: https://github.com/w3c/did-spec-registries/pull/361/files#diff-728e5baeea70198c17ae62acf75cea4ad381276558f8648cbd8ff5e3ca312a97R17

I suggest, we remove it from this PR, but keep the index file you have generated from it, so the PR review can be demonstrated to be functioning correctly.

Reply:
> When we have all these things merged, a new PR from a did method author will have a single JSON file in it, and if that file is valid, the validation will pass, and PR preview will be generated which contains the new entry... there will only be a single file to review in that PR, and that file will be constrained by the CI system, which will ensure it has all required fields.

I'm good with this.